From eb643ec6bd7e22d695418d3d146965c5224e28a0 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:33:13 -0400
Subject: [pci] Add bridge DMA alias quirk for Intel 82801 bridge

Message-id: <20140905023313.497.25670.stgit@gir.stowe>
Patchwork-id: 91381
O-Subject: [RHEL7.1 PATCH 4/7] PCI: Add bridge DMA alias quirk for Intel 82801 bridge
Bugzilla: 1136539
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136539
Upstream Status: 8ab4abbee6df256e7afb7e90a3332aed7453d538

commit 8ab4abbee6df256e7afb7e90a3332aed7453d538
Author: Alex Williamson <alex.williamson@redhat.com>
Date:   Sat Jul 5 15:26:52 2014 -0600

    PCI: Add bridge DMA alias quirk for Intel 82801 bridge

    This bridge sometimes shows up as a root complex device and sometimes as a
    discrete PCIe-to-PCI bridge.  Testing indicates that in the latter case, we
    need to enable the PCIe bridge DMA alias quirk.

    Reported-by: Milos Kaurin <milos.kaurin@gmail.com>
    Tested-by: Milos Kaurin <milos.kaurin@gmail.com>
    Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 1e7253b..815f724 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -3415,6 +3415,8 @@ DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_ASMEDIA, 0x1080,
 DECLARE_PCI_FIXUP_HEADER(0x10e3, 0x8113, quirk_use_pcie_bridge_dma_alias);
 /* ITE 8892, https://bugzilla.kernel.org/show_bug.cgi?id=73551 */
 DECLARE_PCI_FIXUP_HEADER(0x1283, 0x8892, quirk_use_pcie_bridge_dma_alias);
+/* Intel 82801, https://bugzilla.kernel.org/show_bug.cgi?id=44881#c49 */
+DECLARE_PCI_FIXUP_HEADER(0x8086, 0x244e, quirk_use_pcie_bridge_dma_alias);
 
 static struct pci_dev *pci_func_0_dma_source(struct pci_dev *dev)
 {
-- 
1.7.1